### ISSUE TITLE
Session Present Flag is not respected

### ISSUE THREAD

**Issue by lizhanhui** [apache/rocketmq-mqtt#281]
MQTT protocol defines Session Present Flag as follows:
<img width="962" alt="image" src="https://github.com/apache/rocketmq-mqtt/assets/641819/ada42f57-7af0-4168-908e-c27d1192d620">

ConnectHandler writes ConnAck having session-present=false constantly: 

See: 
https://github.com/apache/rocketmq-mqtt/blob/477d478316f95c24bfec0da7f80217dd7d605073/mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/protocol/mqtt/handler/MqttConnectHandler.java#L109

https://github.com/apache/rocketmq-mqtt/blob/477d478316f95c24bfec0da7f80217dd7d605073/mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/protocol/mqtt/facotry/MqttMessageFactory.java#L42

**Comment by DongyuanPan** [comment]
Yes, currently there's no storage for subscription relationships, meaning it doesn't support cleansession=false. Therefore, the default setting is false

**Comment by RockteMQ-AI** [comment]
**Issue Evaluation**

Category: `type/bug` | Status: **Confirmed**

The MQTT 5.0 specification defines the Session Present flag in CONNACK to indicate whether the server has a stored session for the client. `ConnectHandler` always sets `session-present=false`, which violates the protocol when a client reconnects with `Clean Start=false` and the server does have a stored session.

**Severity:** Medium — protocol compliance issue; affects clients that rely on session resumption.

**Suggested Fix:** Check the session store in `ConnectHandler` before building the CONNACK, and set the Session Present flag based on whether a session exists for the client ID.

---
*Automated evaluation by github-manager*

**Comment by RockteMQ-AI** [comment]
**Issue Evaluation**

Category: `type/bug` | Status: **Confirmed**

The Session Present flag not being respected is a protocol compliance issue. MQTT clients rely on this flag to determine whether the broker retained their session state (subscriptions, queued messages). Incorrect flag values break client reconnection logic.

---
*Automated evaluation by github-manager*
